### ISSUE TITLE
som.quantization_error(data) returns "nan"

### ISSUE THREAD

**Issue by TimNagle-McNaughton** [JustGlowing/minisom#51]
Getting "nan" returned when I attempt to run som.quantization_error(data) returns "nan".

----
### Data
I thought the issue may have been related to numpy's [linalg.norm function](https://github.com/numpy/numpy/issues/6128), but I have ensured the data being passed to minisom are a numpy float array:

  | 0 | 1| 2 | 3 | 4 |
-- | -- | -- | -- | -- | -- |
426.0 | -2403.0 | -63.0 | -177.0 | 14.036022 | 361.0
388.0 | -2458.0 | -8.0 | -50.0 | 14.478539 | 359.0
405.0 | -2432.0 | 31.0 | -19.0 | 13.921394 | 360.0
440.0 | -2348.0 | 87.0 | -95.0 | 14.062276 | 361.0
436.0 | -2259.0 | 219.0 | 9.0 | 13.603740 | 357.0

----
### Code
`# set SOM parameters`

`som = MiniSom(x = x,
             y = y,
             input_len = input_length,
             sigma = sigma,
             learning_rate = learning_rate,
             random_seed = random_seed)`

`# train the SOM`

`start_time = time.time()`

`som.train_random(np_data, iterations)`

`elapsed_time = time.time() - start_time`

`print("Training time: {}".format(elapsed_time))`

----

### Solutions attempted
Based on my reading of the quantization_error function:
`    def quantization_error(self, data):`

        `"""Returns the quantization error computed as the average

        distance between each input sample and its best matching unit."""

        self._check_input_len(data)

        error = 0

        for x in data:

            error += fast_norm(x-self._weights[self.winner(x)])

        return error/len(data) `

I've checked my data length (1,932,042), and weights: 

[[[ 0.40056419 -0.70753778  0.19731155  0.36732157 -0.00220417
   -0.40629643]
  [-0.36671443  0.31642472 -0.4018778  -0.49997736  0.22512673
    0.55066391]
  [-0.74021007  0.01819334  0.46649405  0.16791178  0.33090403
   -0.31056323]
  ...

Which are seem to be as expected. 

The only thing I can think of is that my winning nodes are not correct.

Any help much appreciated.

**Comment by JustGlowing** [comment]
hi there,

Here are some suggestions that can solve the issue:

- Make sure that there's no `nan` values in your input data.
- Check that there's no `nan` in the weights of your winning neurons.
- Check the values of `len(np_data)`.

On a separate note, normalize you data before using minisom. Standard scaling is the technique that works in most of the cases.

If you don't solve the issue I'll try to replicate your situation on my machine.

**Comment by TimNagle-McNaughton** [comment]
Normalizing the data worked, and I had some hidden NaN values in my input data from using incorrect "na_values = ____, na_filter = True)" in my read_csv call.

Thanks as always.
